public/src/modules/iconSelect.js

```diff
@@ -41,29 +42,53 @@ define('iconSelect', function() {
 								var iconClass = $('.bootbox .selected').attr('class');
 								var categoryIconClass = $('<div/>').addClass(iconClass).removeClass('fa').removeClass('selected').attr('class');
 
-								el.attr('class', 'fa ' + (doubleSize ? 'fa-2x ' : '') + categoryIconClass);
-								el.val(categoryIconClass);
-								el.attr('value', categoryIconClass);
+								if (categoryIconClass) {
+									el.attr('class', 'fa ' + (doubleSize ? 'fa-2x ' : '') + categoryIconClass);
+									el.val(categoryIconClass);
+									el.attr('value', categoryIconClass);
+								}
 
 								onModified(el);
 							}
 						}
 					}
 				});
 
+			picker.on('show.bs.modal', function() {
+				var modalEl = $(this),
+					searchEl = modalEl.find('input');
+
+				if (selected) {
+					modalEl.find('.' + selected).addClass('selected');
+					searchEl.val(selected.replace('fa-', ''));
+				}
+			}).modal('show');
+
 			picker.on('shown.bs.modal', function() {
 				var modalEl = $(this),
 					searchEl = modalEl.find('input'),
 					icons = modalEl.find('.fa-icons i'),
 					submitEl = modalEl.find('button.btn-primary');
 
+				function changeSelection(newSelection) {
+					modalEl.find('i.selected').removeClass('selected');
+					if (newSelection) {
+						newSelection.addClass('selected');
+					} else if (searchEl.val().length === 0) {
+						if (selected) {
+							modalEl.find('.' + selected).addClass('selected');
+						}
+					} else {
+						modalEl.find('i:visible').first().addClass('selected');
+					}
+				}
+
 				// Focus on the input box
-				searchEl.focus();
+				searchEl.selectRange(0, searchEl.val().length);
```

Comment:
Leaving the input text selection here is actually kind of nice. The visible fade-in effect of the input draws attention to it.